### PR TITLE
Install react with CRA

### PR DESCRIPTION
**Pull request by M-AminAlizadeh**
The things that I did in this PR:

-   Create a new React app called 'hello-react-front-end'
-   Initialize your project with Git.
-   Create the App component with react-router.
-   Create the Greeting component that will display a greeting. Set it up as a route in your App component.
-   Create a store, an action, and a reducer that will connect to the API endpoint to get the random greeting.
-   Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by tugsem** [comment]
## Invalid Code Review Request ❌ 

Hi @M-AminAlizadeh , 

While you made a great effort in this project, I cannot proceed to review your code.

<h2>You have submitted a project, that doesn't have a summary in the description. Pull Request should have a summary describing the introduced changes in more detail.</h2>

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/Greetings.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const Greetings = () => {
+  const greeting = useSelector((state) => state.greeting);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const fetchGreetings = async () => {
+      const response = await fetch('http://127.0.0.1:3000/greetings/random');
+      if (response.ok) {
+        const data = await response.json();
+        dispatch(setGreeting(data.greeting));
```
Comment (LionRouge1) [inline src/components/Greetings.jsx:14]: - [ ] Kindly move the API call functionality into the reducer. So the connection between your front-end and your API is handled by the reducer to follow the project requirements.
Screenshot from 2023-12-23 15-04-55

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @M-AminAlizadeh, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- React app uses the url and displays a random greeting :heavy_check_mark: 
- Create `App` and `Greetings` components :heavy_check_mark: 
- Set up the `Greetings` component as a route in the `App` component :x:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Greeting.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { connect } from 'react-redux';
+import { setGreeting } from '../greetingSlice';
+
+const Greeting = ({ greeting, setGreeting }) => {
+  useEffect(() => {
+    fetch('http://127.0.0.1:3000/greetings/random')
+      .then((response) => response.json())
+      .then((data) => setGreeting(data.greeting))
+      .catch((error) => console.error('Error fetching greeting:', error));
+  }, [setGreeting]);
```
Comment (Ibizugbe) [inline src/components/Greeting.jsx]: Just as the [previous reviewer](https://github.com/M-AminAlizadeh/Hello-react-front-end/pull/1#discussion_r1435623407) has suggested, kindly ensure the `api` call is made from the reducer😉. as this is a requirement for this milestone😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @M-AminAlizadeh 👋🏼👋🏼👋🏼, 

request change


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] React router has been used👍🏼.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tugsem** [comment]
## STATUS: APPROVED 👍🎯  
Hello @M-AminAlizadeh 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️action called to get API message
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**